src/socket/socket.cr

```diff
@@ -32,6 +32,48 @@ class Socket < IO::FileDescriptor
 
     def initialize(@family, @path)
     end
+
+    def initialize(sockaddr : LibC::SockAddrIn | LibC::SockAddrIn6)
+      @family = case sockaddr.family
+      when LibC::AF_INET then "AF_INET"
+      when LibC::AF_INET6 then "AF_INET6"
+      else raise "Unsupported address family"
```

Comment:
Yes, breaking everything everywhere if it is for a better solution is still okay.